Gnoll Wars Bookmark WIP

fivesanon opened the pull request:
# TO DO:
-Gnoll
--More portrait options??????
--Events
---Raiding Events only for gnollish culture group. (vore😳)
-Bloodlines
--Barathen
---Monthly cost for levies and retinues -5%
---Cost for retinues -25%
-Gurubashi
--characters
-Kul Tiras
--Clean up the holdings there, briarback should be tribal
--Extend dynasties

# IMPORTANT TO DO:
-make cb for gnoll war
-fill in no character counties
-find land owned by characters who are dead or havnt been born yet

# HOW TO TEST:
Search for dumb shit in this branch, first mod I've done so yeah...

# CHANGELOG:
-Gnoll War start date
-Gnoll portraits

zumbak04 commented:
@fivesanon hello!

> couldnt figure out how to make the CB give them land.

What CB do you mean?

zumbak04 commented on history/characters/29000_gnoll.txt:
```diff
@@ -808,82 +813,47 @@
-	mother=29079	#Toothy
-	342.2.19={ birth=yes add_trait=creature_gnoll }
-	399.7.12={ death=yes }
 }
 29077={
-	name=Marauder
-	female=yes
-	dynasty=12232
-	dna="hnlldlffddl"
-	culture=gnoll religion=primitive_shamanism
-	father=29072	#Pippin
-	mother=29079	#Toothy
-	347.10.11={ birth=yes add_trait=creature_gnoll }
-	402.1.6={ death=yes }
 }
```
Delete IDs too if you're going to remove them.

fivesanon replied:
I was saving the ids for later when i add more characters to his dynasty.

zumbak04 commented on history/titles/c_atalhakkar.txt:
```diff
@@ -1,5 +1,9 @@
 # d_swamp_of_sorrows
+481.12.16={
+	holder=6039
+	liege=k_atalai
+}
 560.1.1={
 	holder=6035
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6035
```

zumbak04 commented on history/titles/c_bogpaddle.txt:
```diff
@@ -1,3 +1,7 @@
+481.12.16={
+	holder=0
+	liege=k_atalai
+}
 560.1.1={
 	holder=6024
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6024
```

zumbak04 commented on history/titles/c_harborage.txt:
```diff
@@ -1,4 +1,8 @@
 # k_atalai
+481.12.16={
+	holder=0
+	liege=k_atalai
+}
 560.1.1={
 	holder=6005
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6005
```

zumbak04 commented on history/titles/k_redridge.txt:
```diff
@@ -0,0 +1,6 @@
+	law = succ_elective_gavelkind
+}
+499.5.11={holder=29071} #Garfang[12232]
+	capital=33
```
Doesn't work because it's out of brackets.
```suggestion
499.5.11={
	capital=33
}
```

zumbak04 commented on history/titles/k_atalai.txt:
```diff
@@ -1,3 +1,7 @@
+481.12.16={
+	holder=6039
+	liege=k_atalai
+	}
```
Nothing special, just correct indentation.
```suggestion
481.12.16={
	holder=6039
	liege=k_atalai
}
```

zumbak04 commented on history/titles/d_stonewatch.txt:
```diff
@@ -1,3 +1,6 @@
+	law = cognatic_succession
+	law = succ_elective_gavelkind
+}
+499.5.11={holder=29071} #Garfang[12232]
 }
```
Extra bracket.
```suggestion
499.5.11={holder=29071} #Garfang[12232]
```

fivesanon replied:
fixed

zumbak04 commented on history/titles/d_riverpaw.txt:
```diff
@@ -1,6 +1,12 @@
+491.12.29={
+	holder=29206
+}
+507.07.11={
+liege=k_redridge
+}
```
Nothing special, just correct indentation.
```suggestion
507.07.11={
	liege=k_redridge
}
```

zumbak04 commented on history/titles/c_stonard.txt:
```diff
@@ -1,4 +1,8 @@
+481.12.16={
+	holder=0
+	liege=k_atalai
+	}
```
Nothing special, just correct indentation.
```suggestion
}
```

zumbak04 commented on history/titles/c_richgarden.txt:
```diff
@@ -1,27 +1,20 @@
+	law = cognatic_succession
+	law = succ_elective_gavelkind
+}
+507.07.11={
+holder=29194
+liege=k_redridge
 }
```
Nothing special, just correct indentation.
```suggestion
507.07.11={
	holder=29194
	liege=k_redridge
}
```

zumbak04 commented on history/titles/c_lakeshire.txt:
```diff
@@ -1,9 +1,13 @@
 	law = succ_primogeniture
 }
+499.5.11={
+	liege=c_redridge_canyons
```
Count title can't be vassal of another count.

fivesanon replied:
fixed

zumbak04 commented on history/titles/c_marshtide.txt:
```diff
@@ -1,4 +1,8 @@
+481.12.16={
+	holder=0
+	liege=k_atalai
+}
 560.1.1={
 	holder=6020
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6020
```

zumbak04 commented on history/titles/c_north_slough.txt:
```diff
@@ -1,5 +1,9 @@
 # k_atalai
+481.12.16={
+	holder=6039
+	liege=k_atalai
+}
 560.1.1={
 	holder=6016
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6016
```

zumbak04 commented on history/titles/c_sorrowmurk.txt:
```diff
@@ -1,4 +1,8 @@
+481.12.16={
+	holder=0
+	liege=k_atalai
+}
 560.1.1={
 	holder=6028
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6028
```

zumbak04 commented on history/titles/c_south_slough.txt:
```diff
@@ -1,4 +1,8 @@
 # k_atalai
+481.12.16={
+	holder=0
+	liege=k_atalai
+	}
 560.1.1={
 	holder=6010
 	liege=k_atalai
```
The same liege set twice.
```suggestion
	liege=k_atalai
}
560.1.1={
	holder=6010
```

zumbak04 commented on history/titles/c_stagalbog.txt:
```diff
@@ -1,4 +1,9 @@
 560.1.1={
-	holder=6032
+	holder=6040
 	liege=k_atalai
```
The same liege set twice.
```suggestion
```

fivesanon commented:
> @fivesanon hello! Wow, an ambitious feature!
> 
> > couldnt figure out how to make the CB give them land.
> 
> What CB do you mean?

i originally had all this in a seperate mod that i was basically working on for the whole day yesterday, i copied all the folders into the main mod folder and added it to github, but i guess i forgot the common folder lol.

Malicos commented:
Also keep in mind there is work in progress in-house lore being developed for a potential Three Hammers bookmark. There might be some tuning to the dynasties mentioned in the yet unfinished document.

I am linking it for reference:
https://docs.google.com/document/d/1wy2K9Nl9R-2oGEmqxUZTaxkhMdrTKNUJNZNbQAOfuEw/

zumbak04 commented:
> i originally had all this in a seperate mod that i was basically working on for the whole day yesterday, i copied all the folders into the main mod folder and added it to github, but i guess i forgot the common folder lol.

Oh, got it, consider using vanilla `tribal_invasion` CB. It's used by the Aztecs, Hungarians. It will save your time if you don't plan anything special in the gnoll CB.

> I also suggest making a similar documentation for Gnoll Wars.

Not critical in my opinion if you don't plan history tweaks in other regions.

zumbak04 commented:
One more thing, you missed a few of my comments (they're hidden).
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/81725646-35b13580-9497-11ea-87b2-2bd0917acb76.png)
![image](https://user-images.githubusercontent.com/46384992/81725679-3fd33400-9497-11ea-9bc0-ea78ff3b116c.png)

</details>

When you finish, click the cycle button to request my review. I'll check more tomorrow.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/81725739-56798b00-9497-11ea-8dbe-5fa829be2559.png)

</details>

fivesanon commented:
> One more thing, you missed a few of my comments (they're hidden).
> 
> Click to expand
> ![image](https://user-images.githubusercontent.com/46384992/81725646-35b13580-9497-11ea-87b2-2bd0917acb76.png)
> ![image](https://user-images.githubusercontent.com/46384992/81725679-3fd33400-9497-11ea-9bc0-ea78ff3b116c.png)
> 
> When you finish, click the cycle button to request my review. I'll check more tomorrow.
> 
> Click to expand
> ![image](https://user-images.githubusercontent.com/46384992/81725739-56798b00-9497-11ea-8dbe-5fa829be2559.png)

ill fix those rn, also added /common folder so the cb is in there for the war. i wanted the gnolls outside redridge to be tributaries but couldnt figure out how to give them land in the cb so i just made them vassals and made garfang usurp them.

Malicos commented on common/objectives/00_plots.txt:
```diff
@@ -770,8 +770,7 @@ plot_kill_character = {
 		is_untouchable_trigger = no
-		is_humanoid_trigger = yes
-		has_invader_title_trigger = no
+		sits_on_frozen_throne_trigger = no
```
These changes are not present on current master. Have you been merging other branches into your own or basing it on a non-master branch?

Malicos commented on common/objectives/00_plots.txt:
```diff
@@ -1172,8 +1171,7 @@ plot_kill_spouse = {
 		is_untouchable_trigger = no
-		is_humanoid_trigger = yes
-		has_invader_title_trigger = no
+		sits_on_frozen_throne_trigger = no
```
These changes are not present on current master. Have you been merging other branches into your own or basing it on a non-master branch?

Malicos commented on history/characters/29000_gnoll.txt:
```diff
@@ -808,82 +813,47 @@
-	289.4.25={ birth=yes add_trait=creature_gnoll }
-	310.1.1={ add_spouse=29078 }
-	348.2.24={ death=yes }
+	martial=10 diplomacy=12 stewardship=1 intrigue=10 learning=12
```
Too OP. Try balancing his stats around the total number of 24.

Malicos commented on history/characters/29000_gnoll.txt:
```diff
@@ -1519,4 +1488,66 @@
+	name=Razer
+	dynasty=12236
+	culture=gnoll religion=primitive_shamanism
+	martial=9 diplomacy=4 stewardship=6 intrigue=9 learning=2
```
Too OP. Try balancing his stats around the total number of 24.

Malicos commented on history/characters/29000_gnoll.txt:
```diff
@@ -1519,4 +1488,66 @@
+	name=Shredder
+	dynasty=12237
+	culture=gnoll religion=primitive_shamanism
+	martial=2 diplomacy=6 stewardship=11 intrigue=4 learning=4
```
Stewardship too OP. Try balancing his stats around the total number of 24.

Malicos commented on history/characters/29000_gnoll.txt:
```diff
@@ -1519,4 +1488,66 @@
+	name=Teef
+	dynasty=12001
+	culture=gnoll religion=primitive_shamanism
+	martial=9 diplomacy=0 stewardship=1 intrigue=3 learning=0
```
Why some of his stats are 0? Try balancing his stats around the total number of 24.

fivesanon replied:
done

Malicos commented on history/characters/6000_gurubashi.txt:
```diff
@@ -373,7 +374,39 @@
+	name=Drukinjo
+	dynasty=8097
+	culture=gurubashi religion=cult_of_hakkar
+	martial=9 diplomacy=8 stewardship=7 intrigue=7 learning=5
```
Too OP. Try balancing his stats around the total number of 24.

Malicos commented on history/characters/6000_gurubashi.txt:
```diff
@@ -373,7 +374,39 @@
+	name=Sinthal
+	dynasty=8414
+	culture=gurubashi religion=cult_of_hakkar
+	martial=9 diplomacy=4 stewardship=6 intrigue=8 learning=3
```
Too OP, particularly in Martial/Intrigue. Try balancing his stats around the total number of 24.

Malicos commented on localisation/00_wcgw_localisation.csv:
```diff
@@ -0,0 +1,3 @@
+#CODE;ENGLISH;FRENCH;GERMAN;;SPANISH;;;;;;;;;x
+garfang;Blood of [This.GetFirstNameWithNick];;Blut von [This.GetFirstNameWithNick];;;;;;;;;;;x
+garfang_desc;Garfang was a cunning and intelligent gnoll alpha of the Redridge pack, who unified the gnoll tribes and began launching bold attacks on the kingdom of Stormwind, beginning the Gnoll War.
```
Duplicate.

If it is not a submod, I would just put it into `00_zuwc_united_localisation.csv` instead.

fivesanon replied:
done

Malicos commented:
Also, I have checked your fork, and it seems your branch is based off origin's small-improvments-20.
You should never create your branches based on anything else than master.

MemeWizard commented:
> You should never create your branches based on anything else than master.

Nah, it's fine.
I'll just rebase this branch on small-improvments-20.

fivesanon commented:
how do i deal with the merging issue, previously before i started this i tried out several of the early version branches that could be the issue, idk.
also i will fix the stats thank you for letting me know :)

MemeWizard commented:
> how do i deal with the merging issue, previously before i started this i tried out several of the early version branches that could be the issue, idk.

That's not a big problem. Since this branch based on both `small-improvments-19` and `small-improvments-20`, it will be resolved automatically when one of them will be merged into `master` branch.

fivesanon commented:
worked alot, didnt get much done
-started on zul'aman
--Zuljin has dad!! (pog)
--it will be lots of chiefdoms that make up the kingdom of zul'aman, so far 2

-atal'ai
--leader is part of vol dynasty(dynasty of leader during first war start date)
--vassal of stagalbog/sorrowmurk is  of wan dynasty(dynasty of vassal during first war start date)
--independent murlocs on rotten beach, same dynasty as first war

-redridge
--did not fix missing vassals, someone please fix it i have no idea what i did wrong, its probably due to my inexperience

-localisations
--redridge pack

biggish issues
-redridge pack vassals not showing, tried fixing it to no avail
-funky shit going down in lakeshire, alther, and blackrock pass after gnoll wars start date, tried fixing it to no avail
-same counties as above wont show up under the redridge realm when you select ruler, i know i can fix it but im tired and want to sleep

please look at these and fix them if you know whats wrong, i want to get better at this so i dont have issues like this again.

zumbak04 commented:
> -redridge pack vassals not showing, tried fixing it to no avail
> -funky shit going down in lakeshire, alther, and blackrock pass after gnoll wars start date, tried fixing it to no avail

@fivesanon, found bugs there. Be careful with brackets. Read [these basics](https://ck2.paradoxwikis.com/Scripting) to understand how it works.

zumbak04 commented on common/cb_types/gw_cb_types.txt:
```diff
@@ -0,0 +1,63 @@
+# CB for gnoll war
+gnoll_war = {
```
Ooof, it's buggy. Answering your comment...

> Make brightwood pack and riverpaw pack tributaries of redridge pack AND gain/lose land upon gnoll war CB (i dont know how to do this at at all so i need help.

I recommend using `tribal_invasion` in the war and deleting this CB. It's not EU4 so there's no good way to give lands to your allies.

zumbak04 commented on common/landed_titles/gw_landed_titles.txt:
```diff
@@ -0,0 +1,2031 @@
+#####TEMPORARY UNTIL TRIBUTARIES OF REDRIDGE PACK GET LAND UPON CB ENFORCED#####
+e_azeroth = {
```
What `e_azeroth` doing in this file?

zumbak04 replied:
Got it. You want `k_redridge` to be de-jure liege of duchies in the Redriege mountains, right? Move the `k_redridge` declaration to `wc_titular_titles` (we store titular titles there) and add the `de_jure_liege` in history of Redriege duchies.
<details>
<summary>Syntax example from vanilla</summary>
867.1.1={
	de_jure_liege = e_hre
}
</details>

zumbak04 commented on history/characters/34000_murloc.txt:
```diff
@@ -6,7 +6,7 @@
 	culture=murloc religion=water_deities
 	martial=6 diplomacy=6 stewardship=8 intrigue=6 learning=8
-	add_trait=tough_soldier add_trait=gluttonous add_trait=ambitious add_trait=kind add_trait=humble 
+	add_trait=tough_soldier add_trait=gluttonous add_trait=ambitious add_trait=kind add_trait=humble
```
@fivesanon you deleted a lot of white spaces. If you work in Atom, disable it.
https://flight-manual.atom.io/using-atom/sections/editing-and-deleting-text/

fivesanon commented:
I figured out the issues I had before
-All the vassals show up again.
-The dynasties work and you can see all the characters in future start dates like Garfang is Knockout's grandfather.
-History fixed for c_cavefall
-As far as I can tell everything works

I will work on adding new stuff now. Thank you :)

fivesanon commented:
-Fixed garfang bloodline opinions
-Added rockpool murlocs in southern coast of black morass
-Added trolls in black morass, not entirely done yet. trying to make "stories" in every area instead of being generic rulers, the 3 trolls i added there are brothers.

zumbak04 commented on history/titles/c_deihsei.txt:
```diff
@@ -13,4 +17,15 @@
+554.5.24={holder=460001} #Fmgrlgr[6134]
+567.3.4={holder=460000} #Neptool[6134]
+583.1.1={holder=52000} #Cho'Gall
+}
```
The bracket doesn't close anything.
```suggestion
```

zumbak04 commented on history/titles/c_shindun.txt:
```diff
@@ -13,4 +17,15 @@
+554.5.24={holder=460001} #Fmgrlgr[6134]
+567.3.4={holder=460000} #Neptool[6134]
+583.1.1={holder=52000} #Cho'Gall
+}
```
The bracket doesn't close anything.
```suggestion
```

fivesanon commented:
![image](https://user-images.githubusercontent.com/57690273/81952176-009e0380-95bb-11ea-9946-751b32067b85.png)
question, should this character stay here. I assume not but im not sure?

zumbak04 commented:
> question, should this character stay here. I assume not but im not sure?

Ooof, don't know, he may stay, but better test it.

zumbak04 commented:
@fivesanon, also, keep images in [spoilers](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/Text-Tricks-on-GitHub#spoilers) to save space... and Earth.

fivesanon commented:
-Riverpaw no longer shows as a vassal of Redridge Pack after the gnoll wars.

-Black morass finished
--Necromancer kobold ruler in c_dead_ravine, he do be vibin
--Stupid dumb troll in d_tainted_forest.
--Black morass is split between loa and hakkari (and nerglish and candle :) )
--Extended date at which Aeonus owns c_dark_portal

fivesanon commented:
made a bookmark image, tried to get it to work, nothing happened. idk what i did wrong, can someone help??

MemeWizard commented:
@fivesanon, when you add image, you need to create graphical assets too. For graphical assets of bookmark images, check `Guardians of Azeroth\interface\wc_other_pictures.gfx` file.

fivesanon commented:
I have made 4 variants of gnolls as well as redid the undead gnoll portrait to fit with the new variants. I could also probably add eye color because thats not that hard to do, but this is it so far.

<details>
<summary>Gnoll Portraits</summary>

![image](https://user-images.githubusercontent.com/57690273/82176958-0e8aa780-988d-11ea-99f0-32258157b637.png)
![image](https://user-images.githubusercontent.com/57690273/82176970-15b1b580-988d-11ea-97f0-8fd7342b28d6.png)

</details>

Also I am working on an event right now called the "skinning festival" and for now ive copy pasted blots from base game, essentially its an intrigue event where you bring all your friends and vassals together, you drag your prisoners out into the kitchen, and you skin them, chop them up, and cook them for a big feast the next day! Fun! :)

<details>
<summary>Skinning Festival</summary>

![20200517190205_1](https://user-images.githubusercontent.com/57690273/82177528-902f0500-988e-11ea-9046-987578fd4330.jpg)
![20200517223403_1](https://user-images.githubusercontent.com/57690273/82177529-90c79b80-988e-11ea-98c6-157c60348534.jpg)

</details>

Im going to bed now.

zumbak04 commented:
> I have made 4 variants of gnolls

> Also I am working on an event right now called the "skinning festival"

@fivesanon sounds fun, but better make such things on separate branches. Big branches take more time to review. So if a feature can be separated, it should be separated.
So could you move them?

zumbak04 commented:
@fivesanon you missed 3 of my comments above. Like this:
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/82323314-ef089180-99e8-11ea-8e14-0487eeba9117.png)

</details>
50% of the changes are whitespace changes.

zumbak04 commented:
Also, I recommend reading [this](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/How-to-be-Tinker-(Developer)) since you're doing a big feature.

fivesanon commented:
> @fivesanon you missed 3 of my comments above. Like this:
> 
> Click to expand
> ![image](https://user-images.githubusercontent.com/46384992/82323314-ef089180-99e8-11ea-8e14-0487eeba9117.png)
> 
> 50% of the changes are whitespace changes.

sorry, thought i got the whitespace thing but ig not, think i fixed it



> Also, I recommend reading [this](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/How-to-be-Tinker-(Developer)) since you're doing a big feature.

Thank you, I was not aware of the rules and some features on github. should i @ you for questions???

zumbak04 commented:
@fivesanon I recommend checking what you added so far, I've seen many reverts.
Also, I recommend handling [TortoiseGit](https://tortoisegit.org/).
It allows such thing as [comparing branches](https://tortoisegit.org/docs/tortoisegit/tgit-dug-diff.html) (like your and `small-improvments-19`).

> If you want to see the difference between a particular revision and your working tree, use the Log dialog, select the revision of interest, then select Compare with working tree from the context menu (cf. the section called “Log Dialog”).

That's what I used to find reverts.

fivesanon commented:
> @fivesanon I recommend checking what you added so far, I've seen many reverts.
> Also, I recommend handling [TortoiseGit](https://tortoisegit.org/).
> It allows such thing as [comparing branches](https://tortoisegit.org/docs/tortoisegit/tgit-dug-diff.html) (like your and `small-improvements-19`).
> P.S. `small-improvements-19` must be base because you branch share improvements' content.
> 
> > If you want to see the difference between a particular revision and your working tree, use the Log dialog, select the revision of interest, then select Compare with working tree from the context menu (cf. the section called “Log Dialog”).
> 
> That's what I used to find reverts.

got it 👍

zumbak04 commented:
Deleting whitespaces again in cecb79aa2b6e41c5f2648fa30d7edd3346005d28.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/83351159-a6d76080-a352-11ea-9003-ff45cecb8901.png)

</details>

zumbak04 commented:
These events can be deleted since they're moved to another branch, right?
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/83351190-e0a86700-a352-11ea-9453-8e032cc6e8ac.png)

</details>

zumbak04 commented:
My bad, didn't know I'll close this PR by deleting the small improvements.

fivesanon commented:
> These events can be deleted since they're moved to another branch, right?
yes they can.
also i will start working again tomorrow, maybe even today.

zumbak04 commented on history/titles/c_venomweb.txt:
```diff
@@ -2,8 +2,25 @@
+466.07.08={holder=29290} #Snarltooth[12238]
+482.06.05={holder=29292} #Mogvo[12238]
+515.08.21={holder=29215} #Throbal[12238]
 }
```
Borken bracket.
```suggestion
515.08.21={holder=29215} #Throbal[12238]
```

Malicos commented:
@fivesanon do you intend to continue working on this anytime soon?

Malicos commented:
[moved to another branch]
